core/modules/shrscan/Director.h

```diff
@@ -0,0 +1,315 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2014-2015 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_QSERV_SSCAN_DIRECTOR_H
+#define LSST_QSERV_SSCAN_DIRECTOR_H
+
+// System headers
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace lsst {
+namespace qserv {
+namespace shrscan {
+
+//-----------------------------------------------------------------------------
+//! @brief Describe a required table for a shared scan query.
+//-----------------------------------------------------------------------------
+
```

Comment:
This comment is for every line from here to EOF - it needs some serious work to correct indentation, all in-class code must be indented, comments included. See 6-5 in https://confluence.lsstcorp.org/pages/viewpage.action?pageId=16908737